nupic/data/record_stream.py

```diff
@@ -25,178 +25,271 @@
 import datetime
 
 
+from nupic.data.fieldmeta import FieldMetaSpecial
 
-class RecordStreamIface(object):
-  """This is the interface for the record input/output storage classes."""
 
-  __metaclass__ = ABCMeta
 
+def _getFieldIndexBySpecial(fields, special):
+  """ Return index of the field matching the field meta special value.
+  :param fields: sequence of nupic.data.fieldmeta.FieldMetaInfo objects
+    representing the fields of a stream
+  :param special: one of the special field attribute values from
+    nupic.data.fieldmeta.FieldMetaSpecial
+  :returns: first zero-based index of the field tagged with the target field
+    meta special attribute; None if no such field
+  """
+  for i, field in enumerate(fields):
```

Comment:
@oxtopus, I added a check to make sure that `fields` arg evaluates to true, which covers the non-empty case. I leave the rest up to duck typing. I am hesitant about the check that you suggested, as it is conventional in python to use the "duck typing" style. By explicitly checking for `list` and `tuple`, we're reducing the generality of the implementation, since anything that implements the sequence protocol should be an acceptable container in this case. Second, you would need a pretty unlikely confluence of getting the wrong container and having "special" as the attribute of each of its elements. I don't believe that we have a problem here that needs to be solved.